Update so newlines will be filtered from logging

GalateaTom opened the pull request:
Change does similar job to https://github.com/galatea-associates/fuse-starter-java/pull/244
Unlike @Gala-Sortino 's change this method will not filter out \t but it will still filter out newlines.  It's more concise and should be safer as it runs on every log line, not just those when we call encode.
Change been tested on hardcoded logs to prove that newlines are escaped, although I have not been able to test this against input actually provided to the RESTful service because those characters are escaped before they get to the logger (so newlines weren't appearing in logs even before I made this change).
Happy to discuss what best solution is.

GalateaTom committed:
Update so newlines will be filtered from logging

GalateaFarhaad commented on pom.xml:
```diff
@@ -58,6 +59,37 @@
+				<exclusion>
+					<groupId>org.apache.logging.log4j</groupId>
+					<artifactId>log4j-slf4j-impl</artifactId>
+				</exclusion>
```
I don't think it's necessary to exclude these transitive dependencies. Adding them as explicit dependencies with explicit versions should be sufficient to override the transitive versions.

GalateaTom replied:
Ok, I was wondering if this was necessary or not.  Had a read through http://maven.apache.org/guides/introduction/introduction-to-dependency-mechanism.html and you're right, it's enough that we're stating the versions later on.

GalateaTom commented:
I've run few tests using Postman and confirm that using that we can pass in line breaks and they are not included in the logs:
![Logging Line Breaks](https://user-images.githubusercontent.com/24912859/62383836-af9bb200-b548-11e9-9102-59b007996fc6.png)

GalateaTom committed:
Remove Unnecessary Exclusions

The dependency versions stated elsewhere in the pom will override the transitive dependencies automatically, so no need to exclude those transitive dependencies

GalateaTom committed:
Revert "Remove Unnecessary Exclusions"

This reverts commit 22620d8fa1fcb19690fffc237786154b190fa343.

GalateaTom committed:
Revert "Update so newlines will be filtered from logging"

This reverts commit 258427e0a7c6512797c545feb5c42d16f26f5551.

GalateaTom committed:
Add temporary fix to remove special characters from request params

GalateaTom committed:
Break up long comments to fit 100 character line limit

GalateaTom committed:
Make changes recommended by SonarCloud

GalateaTom committed:
Merge branch 'develop' into feature/issue-243-protect-against-log-forging-TomEd

GalateaEric commented on src/main/java/org/galatea/starter/entrypoint/BaseRestController.java:
```diff
@@ -16,9 +18,21 @@
+      try {
+        cleanedRequestId = UriUtils.encode(requestId, "ISO-8859-1");
+        log.info("Request received.  Cleaned id: {}", cleanedRequestId);
+      } catch (UnsupportedEncodingException e) {
```
have a look at: https://projectlombok.org/features/SneakyThrows for this case.  In particular have a look at the 2nd "common use case".

GalateaTom replied:
Change made

GalateaEric commented on src/main/java/org/galatea/starter/entrypoint/BaseRestController.java:
```diff
@@ -16,9 +18,21 @@
-      Tracer.setExternalRequestId(requestId);
+      String cleanedRequestId;
+      try {
+        cleanedRequestId = UriUtils.encode(requestId, "ISO-8859-1");
```
let's stick with just UTF-8 for now.  UTF-8 apparently must be available in the JVM.

Further reading: https://www.garron.me/en/linux/charset-encoding-html-change.html

GalateaTom replied:
Change made

GalateaTom committed:
Code Review Changes
